File: core/BaseballGame/BaseballGame.java
package com.dbc;

public class BaseballGame {
    public int calPoints(String[] ops) {
        int[] scores = new int[ops.length];
        scores[0] = Integer.parseInt(ops[0]);
        int res = scores[0], cur = 1;
        for (int i = 1; i < ops.length; i++) {
            if (ops[i].equals("+")) scores[cur] = scores[cur - 1] + scores[cur - 2];
            else if (ops[i].equals("D")) scores[cur] = scores[cur - 1] * 2;
            else if (ops[i].equals("C")) {
                cur--;
                res -= scores[cur];
                continue;
            } else {
                scores[cur] = Integer.parseInt(ops[i]);
            }
            res += scores[cur];
            cur++;
        }
        return res;
    }
}

File: core/BaseballGame/BaseballGame.py
from typing import List

def calPoints(ops: List[str]) -> int:
    scores = [0] * len(ops)
    res, cur, length = int(ops[0]), 1, len(ops)
    scores[0] = res
    for i in range(1, length):
        if ops[i] == "+":
            scores[cur] = scores[cur - 1] + scores[cur - 2]
        elif ops[i] == "D":
            scores[cur] = scores[cur - 1] * 2
        elif ops[i] == "C":
            cur -= 1
            res -= scores[cur]
            continue
        else:
            scores[cur] = int(ops[i])

        res += scores[cur]
        cur += 1

    return res


if __name__ == '__main__':
    print(calPoints(["5","-2","4","C","D","9","+","+"]))
